pkg/controller/machinedeployment/rolling_test.go

```diff
@@ -0,0 +1,176 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package machinedeployment
+
+import (
+	"context"
+	"errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/cluster-api/pkg/apis/cluster/common"
+	"sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+	"testing"
+)
+
+type myclient struct{
```

Comment:
We could use controller-runtime fake client to achieve something similar without the need of mocking the interface youself

Reply:
Ah, I wasn't clear, this isn't really meant to merge, it was mostly to show off the bug i believe I've found. it's in a messy state but it does show that we're getting a MachineSet with 1 replica when there clearly should be 3 replicas or no new machine set? i'm not even sure what the expected behavior is.

Let me clarify this point in the description.